src/Microsoft.ML.TorchSharp/Extensions/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.ML.TorchSharp.Extensions
+{
+    internal static class IEnumerableExtensions
+    {
+        public static string ToString<T>(this IEnumerable<T> enumerable, bool verbose = true)
+        {
+            return verbose
+                ? $"[{string.Join(", ", enumerable.Select(e => e.ToString()))}]"
+                : enumerable?.ToString() ?? string.Empty;
+        }
+
+        public static T ArgMin<T>(this IEnumerable<T> source, Func<T, int> getValue)
+        {
+            var keys = source.ToList();     // avoid enumerate twice
+            var values = keys.Select(getValue);
+            var (minSource, minValue) = keys.Zip(values, (first, second) => (first, second)).Aggregate((min, x) => min.second <= x.second ? min : x);
+            return minValue < int.MaxValue ? minSource : default;
```

Comment:
>minValue < int.MaxValue [](http://example.com/codeflow?start=19&length=23)

when this condition can be false? minValue is int and cannot be greater than int.MaxValue.

Reply:
Yeah, not sure why that was done that way. Fixed it now.

Reply:
Ok, reverted to original code. the case where `minValue = int.MaxValue` makes a critical difference to the code itself.